lib/logstash/inputs/log4j.rb

```diff
@@ -59,7 +57,25 @@ def register
     @logger.info("Log4j input")
   end # def register
 
-  public
+  def run(output_queue)
+    if server?
+      while !stop?
+        Thread.start(@server_socket.accept) do |s|
+          s.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
```

Comment:
@guyboertje Nice point!

Instead of doing this on every loop we could probably add this mixin only once at the register method, will would stop jruby from doing his cache.

class TCPSocket
end